src/pages/CouponPage.vue

```diff
@@ -0,0 +1,86 @@
+<template>
+  <div class="coupon-page">
+    <div class="search-container">
+      <div class="search-core">
+        <input type="text" placeholder="Type here..." class="txt-search" v-model="search" @change="searchMethod">
+        <input type="submit" class="btn-search">
+      </div>
+    </div>
+    <div class="box" v-for="deal in categories" :key="deal.Id">
+      <div class="box-header">
+        <div class="header-container">
+          <div class="header-logo">
+            <img :src="deal.Company.Photo.Url" alt="" v-if="deal.Company.Photo != null">
```

Comment:
```suggestion
            <img :src="deal.Company.Photo.Url" alt="" v-if="deal.Company.Photo">
```
Shorter condition to check whether the data is `null` or not